subsys/bluetooth/host/shell/bt_shell_private.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __BT_SHELL_PRIVATE_H
+#define __BT_SHELL_PRIVATE_H
+
+#include <zephyr/shell/shell.h>
+
+/**
+ * @brief printf-like function which sends formatted data stream to the shell.
+ * (Bluetooth context specific)
+ *
+ * This function can be used from the command handler or from threads, but not
+ * from an interrupt context.
+ *
+ * @param[in] color Printed text color.
+ * @param[in] fmt   Format string.
+ * @param[in] ...   List of parameters to print.
+ */
+__printf_like(2, 3) void bt_shell_fprintf(enum shell_vt100_color color,
+					  const char *fmt, ...);
+
+/**
+ * @brief printf-like function which sends formatted data stream to the shell.
+ * (Bluetooth context specific)
+ *
+ * This function can be used from the command handler or from threads, but not
+ * from an interrupt context.
+ *
+ * @param[in] fmt   Format string.
+ * @param[in] ...   List of parameters to print.
+ */
+__printf_like(1, 2) void bt_shell_fprintf_info(const char *fmt, ...);
+__printf_like(1, 2) void bt_shell_fprintf_print(const char *fmt, ...);
+__printf_like(1, 2) void bt_shell_fprintf_warn(const char *fmt, ...);
+__printf_like(1, 2) void bt_shell_fprintf_error(const char *fmt, ...);
+
+/**
+ * @brief Print data in hexadecimal format.
+ * (Bluetooth context specific)
+ *
+ * @param[in] data Pointer to data.
+ * @param[in] len  Length of data.
+ */
+void bt_shell_hexdump(const uint8_t *data, size_t len);
+
+/**
+ * @brief Prints the current command help.
+ * (Bluetooth context specific)
+ *
+ * Function will print a help string with: the currently entered command
+ * and subcommands (if they exist).
+ */
+void bt_shell_help(void);
```

Comment:
`bt_shell_help` should only ever use the shell that tried to do something. I see that we have a case where `cmd_connect_le` is called from a callback. 

In that case (where `sh == NULL`), I would just not print this at all :) It's pretty useless in a callback (and with `wall`), since it related to a specific command.

Reply:
So, `bt_shell_help()` shouldn't be implemented, right? :)

Reply:
I would say so, yeah. It's not useful to wall user errors :)